actions/views/drafts.ts

```diff
@@ -28,15 +29,36 @@ type Draft = {
 }
 
 export function getDrafts(teamId: string) {
-    return async (dispatch: DispatchFunc) => {
-        let drafts: ServerDraft[] = [];
-        drafts = await Client4.getUserDrafts(teamId);
-        const actions = (drafts || []).map((draft) => {
-            const {key, value} = transformServerDraft(draft);
-            return setGlobalItem(key, value);
+    const getLocalDrafts = makeGetDrafts();
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState() as GlobalState;
+
+        let serverDrafts: Draft[] = [];
+        if (syncedDraftsAreAllowedAndEnabled(state)) {
+            try {
+                serverDrafts = (await Client4.getUserDrafts(teamId)).map((draft) => transformServerDraft(draft));
+            } catch (error) {
+                return {data: false, error};
+            }
+        }
+
+        const localDrafts = getLocalDrafts(state);
+        const drafts = [...serverDrafts, ...localDrafts];
+
+        const draftsMap = new Map(drafts.map((draft) => [draft.key, draft]));
+        drafts.forEach((draft) => {
+            const currentDraft = draftsMap.get(draft.key);
+            if (currentDraft && draft.timestamp > currentDraft.timestamp) {
+                draftsMap.set(draft.key, draft);
+            }
+        });
+
+        const actions = Array.from(draftsMap).map(([key, draft]) => {
+            return setGlobalItem(key, draft.value);
         });
```

Comment:
Are we going to also save the local drafts that prevail the server ones (because they either 1. don't exist in the server, 2. are newer than the server ones) back on the server?

Reply:
@koox00 So right now we only save the drafts to the server when the user interacts with them. To save the local drafts that prevail back to the server we'd need to add a new endpoint to handle the upsert of multiple drafts (to avoid making multiple requests). Happy to create a ticket and do that in a separate PR.

Reply:
Ticket created here: https://mattermost.atlassian.net/browse/MM-48992